apps/OboeTester/app/src/main/cpp/analyzer/BaseSineAnalyzer.h

```diff
@@ -145,8 +145,8 @@ class BaseSineAnalyzer : public LoopbackProcessor {
     bool transformSample(float sample, float referencePhase) {
         // Track incoming signal and slowly adjust magnitude to account
         // for drift in the DRC or AGC.
-        mSinAccumulator += sample * sinf(referencePhase);
-        mCosAccumulator += sample * cosf(referencePhase);
+        mSinAccumulator += static_cast<double>(sample) * static_cast<double>(sinf(referencePhase));
```

Comment:
You probably only need to cast one operand. The other will get promoted automatically.